### PR TITLE
Fix name matching for Rust functions

### PR DESCRIPTION
**Pull request by keram88**
This improves matching for Rust function names by requiring they match the syntax of Rust hashed function names or requiring an exact match.

### CODE REVIEW

**Review by shaobo-he — APPROVED** [review]
LGTM.

File: lib/smack/RustFixes.cpp [hunk 1]
```diff
@@ -18,6 +18,17 @@ namespace smack {
 
 using namespace llvm;
 
+bool isRustNameMatch(StringRef search, StringRef name) {
```
Comment (shaobo-he) [inline lib/smack/RustFixes.cpp:21]: Actually this can be written as `return a || b`.

Comment (keram88) [reply lib/smack/RustFixes.cpp:21]: I thought it would be less readable, but I think this is an improvement.

**Review by zvonimir — APPROVED** [review]
Looks brittle, but let's merge it in.

**Comment by zvonimir** [comment]
@keram88 This has conflicts with develop. Could you please resolve those? Thanks!

**Pull request merged**